spec/mem/tree_borrows/memory.md

```diff
@@ -89,10 +89,13 @@ impl<T: Target> TreeBorrowsMemory<T> {
         protected: Protected,
         frame_extra: &mut TreeBorrowsFrameExtra,
     ) -> Result<Pointer<TreeBorrowsProvenance>> {
-        let Some(((alloc_id, parent_path), offset)) = self.check_ptr(ptr, pointee_size)? else {
-            return ret(ptr);
+        let Some((alloc_id, parent_path)) = ptr.provenance else {
+            // An invalid pointer.
+            throw_ub!("Tree Borrows: reborrowing pointer without provenance");
```

Comment:
I don't think this should be UB... how can it be that the tests still pass?^^

This should just do nothing and return the pointer unchanged. Please also add a test for this case, something that creates a `*mut ()` via `ptr::without_provenance_mut` and then turns that into a reference and passes that to a function.